### DISCUSSION TITLE
"Use contexts"

### DISCUSSION THREAD

**Discussion by glaucocustodio** [evenbetterspecs/evenbetterspecs.github.io#3] [General]
What are your thoughts on https://evenbetterspecs.github.io/#use-contexts?

**Comment by asavageiv** [comment]
I used to do this a lot but now less because it makes things less searchable. When I'm copying from a test string like "UsersController create when there is no password returns an error" I have to be very careful about what I copy to find and search the test. The more contexts the harder it is.

**Reply by glaucocustodio** [reply]
how do you test many use cases then?

**Comment by asavageiv** [comment]
I have a top level describe for the class, a describe for the method, and
then I have `it` statements within that spelling out each case in readable
English. It works well with avoiding before and let too

**Reply by glaucocustodio** [reply]
that's an option too, but we find using context more handy

**Reply by d4be4st** [reply]
The main issue with using context is that when you have multiple `it` blocks and you context description exits the screen. When a test fails, and you enter the file you do not have all the information. 
Especially if you allow multiple nested contexts.
